### ISSUE TITLE
Output Meta for powershell resource

### ISSUE THREAD

**Issue by cjbischoff** [inspec/inspec#2492]
### Description

Is it possible to add metadata to a control output; specifically for PowerShell resource checks? When a kitchen verify is performed the output of a successful check is limited to the resource and setting

### InSpec and Platform Version

```
❯❯❯ inspec --version
1.49.2
```

### Replication Case

.kitchen.yml

```
    verifier:
      name: inspec
      format: doc
      inspec_tests:
        - test/integration/default/inspec
```

**This is the control**

```
control 'Virtual Memory Page' do
  impact 0.1
    title 'Ensure Set virtual memory page file size to auto managed'
    desc  'Ensure Set virtual memory page file size to auto managed'
    script = <<-EOH
      (Get-WmiObject -Class Win32_ComputerSystem -EnableAllPrivileges).AutomaticManagedPageFile
    EOH
    describe powershell(script) do
      its('stdout') { should eq "True\r\n" }
    end
end
```
**This is the output from a kitchen verify**

```
Powershell
  stdout
    should eq "True\r\n"
```

### Possible Solutions

Have the ability to out the `desc` tag in the success?

### Stacktrace

NA

Aha! Link: https://chef.aha.io/features/SH-380

**Comment by adamleff** [comment]
Hi @cjbischoff!

If you remove the `format: doc` from your `verifier` section, you will get the InSpec CLI formatted output which includes the control name and ID so you at least know what control the Powershell resource was tested in.

If you want more control over the output, you can switch some tests to use `expect` syntax. See here for more: https://www.inspec.io/docs/reference/profiles/#should-vs-expect-syntax

I'll leave this issue as a feature request for the team to determine if there's a graceful way for us to allow users to add additional control metadata to the output. But I'd recommend removing the `doc` formatter first and seeing if that helps you!

**Comment by cjbischoff** [comment]
Hey @adamleff 

Thank you. I tested without the `doc` setting and that works well for tests that pass, but tests that fail have limited output (with `doc` the failed tests had more meta), but I can definitely work with this - thank you again

I will check out the link

**Comment by adamleff** [comment]
> that works well for tests that pass, but tests that fail have limited output

well, that's certainly not intended! I thought our output was pretty verbose when it came to failures. Can you include an example of a failure from both `doc` and no-`doc` so we can see and potentially address? Obtuseness is never expected. :)

**Comment by miah** [comment]
__What’s happening? Why was this issue closed?__
This issue was closed due to some much needed review of legacy issues or issues that were spawned in older versions of InSpec, i.e. < v3.

__Why do I care?__
You would care about this if this was an issue that you are still seeing and/or feel needs to be addressed in the current version of InSpec.

__What do I need to do?__
If this issue is no longer important, no further action is necessary. However, if you think this is something that should be addressed, please open a new issue and refer to the original issue in the description.

**Comment by dslotter** [comment]
This is still a thorn for some of us. Please consider fixing this. I have a simple (crude) fix for powershell.rb source file:

In initialize method, add:
`@script = script`

In powershell.rb, change the to_s method to something like this:

`"Powershell#{@script}"`

This is obviously not ideal, because if the Powershell script is not a one-liner, you'll get undesired output.

**Comment by jerryaldrichiii** [comment]
I'll reopen this based on our discussion in Slack. I'd like the team to weigh in on if this is what we want.

I'm in favor of it, but not sure about the aesthetic concerns. Maybe gate it behind a flag of some sort?

**Comment by kekaichinose** [comment]
This ask could be solved with the custom reporter plugin that we have included as a part of #3667. Closing this out here and will track the effort there.

**Comment by mcascone** [comment]
@adamleff, cc: @dslotter 
> If you want more control over the output, you can switch some tests to use `expect` syntax. See here for more: [inspec.io/docs/reference/profiles/#should-vs-expect-syntax](https://www.inspec.io/docs/reference/profiles/#should-vs-expect-syntax)

This is really great! I was able to use this to change my tests from: 
```ruby
describe powershell(pools_script) do
  its('stdout') { should eq '' }
end

> [PASS]  Powershell stdout is expected to eq ""
```

To:
```ruby
describe 'All My App Pools' do
  subject { powershell(pools_script).stdout }
  it 'should be started' do
    expect(subject).to(eq '')  
  end
end

> [PASS]  All My App Pools should be started
```

I also get the same result from:
```ruby
describe 'All my App Pools' do
  it 'should be started' do
    expect(powershell(pools_script).stdout).to(eq '')  
  end
end
```
